Fails to build on macOS

I'm not entirely clear on the intent to support macOS, but I noticed that things like cosmic-text require this crate, and that feels a bit more general purpose than just cosmic.

Currently, when building on macOS, you get:

```
error[E0308]: mismatched types
    --> /Users/sophia/.cargo/git/checkouts/cosmic-files-1a0f61508ace3cc6/5732a8d/src/operation/mod.rs:1053:28
     |
1053 |                 return Err("Restoring from trash is not supported on macos".to_string());
     |                        --- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `OperationError`, found `String`
     |                        |
     |                        arguments to this enum variant are incorrect
     |
help: the type constructed contains `std::string::String` due to the type of the argument passed
    --> /Users/sophia/.cargo/git/checkouts/cosmic-files-1a0f61508ace3cc6/5732a8d/src/operation/mod.rs:1053:24
     |
1053 |                 return Err("Restoring from trash is not supported on macos".to_string());
     |                        ^^^^------------------------------------------------------------^
     |                            |
     |                            this argument influences the type of `Err`
note: tuple variant defined here
    --> /Users/sophia/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/result.rs:537:5
     |
537  |     Err(#[stable(feature = "rust1", since = "1.0.0")] E),
     |
```

--- a/src/operation/mod.rs
+++ b/src/operation/mod.rs
@@ -1070,7 +1070,7 @@ impl Operation {
             #[cfg(target_os = "macos")]
             Self::Restore { .. } => {
                 // TODO: add support for macos
-                return Err("Restoring from trash is not supported on macos".to_string());
+                return OperationError::from_str("Restoring from trash is not supported on macos");
             }
             #[cfg(not(target_os = "macos"))]
             Self::Restore { items } => {

